liquigraph-core/src/main/java/org/liquigraph/core/io/GraphJdbcConnector.java

```diff
@@ -10,12 +10,27 @@
 
 public class GraphJdbcConnector implements LiquigraphJdbcConnector {
 
+    private final LockGuardian lockGuardian;
+
+    public GraphJdbcConnector(LockGuardian lockGuardian) {
+        this.lockGuardian = lockGuardian;
+    }
+
+    /**
+     * Acquires a new connection to the configured instance
+     * and tries to lock it (fail-fast).
+     *
+     * @see {@link LockGuardian}
+     * @param configuration Liquigraph settings
+     * @return JDBC connection
+     */
     @Override
     public final Connection connect(Configuration configuration) {
         try {
             Class.forName("org.neo4j.jdbc.Driver");
             Connection connection = DriverManager.getConnection(makeUri(configuration));
             connection.setAutoCommit(false);
+            lockGuardian.lock(connection);
```

Comment:
That's one way on doing it, but the problem is then that you depend on the JVM exiting to clean the lock.

I'd keep the shutdown hook to guard against bad usage, but I think it'd be more versatile by actually wrapping the `Connection` and overriding `close()` to remove the lock.

And now that I think of it, the shutdown hook won't work at all _if_ the connection has already been closed. One more reason to intercept the closing and remove the lock then :).